src/neard_process.rs

```diff
@@ -118,6 +124,48 @@ impl NeardProcess {
     pub async fn wait(&mut self) -> std::result::Result<ExitStatus, std::io::Error> {
         self.process.wait().await
     }
+
+    /// Get Pid of neard
+    pub fn pid(&self) -> Pid {
+        // FIXME: correctly handle unwrap here
+        let pid: i32 = self.process.id().unwrap().try_into().unwrap();
+        Pid::from_raw(pid)
+    }
+
+    /// Update dynamic config
+    /// NOTE: currently only expected shutdown in the config, so input parameter is only expected_shutdown
+    pub async fn update_dynamic_config(
+        pid: Pid,
+        dyn_config_path: &PathBuf,
+        expected_shutdown: BlockHeight,
+    ) -> Result<()> {
+        force_unlink(&dyn_config_path).context("failed to remove previous dynamic config")?;
+        let mut file = File::create(&dyn_config_path).await?;
+        file.write_all(format!("{{\"expected_shutdown\": {expected_shutdown:}}}").as_bytes())
+            .await?;
```

Comment:
Not sure about the exact API, but you get the idea:

```suggestion
        let dynamicConfig = json!({
          "expected_shutdown": expected_shutdown
        });
        file.write_all(dynamicConfig.to_string().as_bytes()).await?;
```